game.py

```diff
@@ -40,33 +40,37 @@ def __init__(self, mapa):
         self._dir = None
         self._map = mapa
 
+    def __reset_rope(self):
+        self._pos = []
+        self._dir = None
+        return
+
     @property
     def stretched(self):
         return self._pos != []
 
     def to_dict(self):
         return {"dir": self._dir, "pos": self._pos}
 
-    def shoot(self, pos, direction, _rocks):
+    def shoot(self, pos, direction, _rocks, _enemies):
         if self._dir and direction != self._dir:
-            self._pos = []  # reset rope because digdug changed direction
-            self._dir = None
-            return
+            return self.__reset_rope()  # reset rope because digdug changed direction
 
         if len(self._pos) > 0:
             new_pos = self._map.calc_pos(self._pos[-1], direction, traverse=False)
         else:
             new_pos = self._map.calc_pos(pos, direction, traverse=False)
 
         if new_pos in [r.pos for r in _rocks]:  # we hit a rock
-            self._pos = []
-            self._dir = None
-            return
+            return self.__reset_rope()
 
         if new_pos in self._pos:  # we hit a wall
-            self._pos = []
-            self._dir = None
-            return
+            return self.__reset_rope()
+        
+        for e in _enemies:  # rope caught fire
+            if type(e) == Fygar and e.fire and self._pos in e.fire:
```

Comment:
```suggestion
            if isinstance(e,Fygar) and e.fire and self._pos in e.fire:
```

Reply:
self._pos é uma lista... 

`self._pos in e.fire` não me parece que funcione

>>> a = [1,2,3]
>>> b = [2,3,4]
>>> a in b
False

Reply:
devia fazer reset para qq posição de `rope` intersectada com qq posição de `fire`

Reply:
fiz esta sugestão de usar o `isinstance` não aceitas....

Reply:
Desculpe professor, eu estou a aprender a usar estas funcionalidades do git ;) aceito sim